Better explain Ruby version requirements (#833)
Shopify/vscode-ruby-lsp: README.md

"rubyLsp.formatter": "auto"
```

#### Ruby version requirement

> By default, the Ruby LSP always uses the current project's Ruby version and bundle. This allows the LSP to find the

Reviewer: ```suggestion
By default, the Ruby LSP uses the current project's Ruby version and bundle. This allows the LSP to find the
```

> correct gem versions automatically to index and to format the code matching behavior on CI.

Reviewer: ```suggestion
correct gem versions automatically so that index and formatting behaviour is consistent across all development and test environments.
```

Author: I don't think this says exactly what I meant. Indexing doesn't happen in CI, it's just formatting that needs to match it.

Reviewer: Ok, how about:

> By default, the Ruby LSP uses the current project's Ruby version and bundle. This allows the LSP to index the correct gem versions, and to ensure formatting behaviour is consistent across all development and test environments.

> The Ruby LSP and its main dependency Prism (the new Ruby parser) both follow the same policy, which is to support only

Reviewer: ```suggestion
The Ruby LSP and its main dependency [Prism](https://github.com/ruby/prism) (the new Ruby parser) both follow the same policy, which is to support only
```

Ruby versions that are not beyond their end-of-life.

If you're working on a project with an older Ruby version, it might be possible to install older versions of the server
gem to get support for older rubies, but that might also involve using older versions of the VS Code extension - since
some functionality requires implementations in both client and server.

> The other alternative is to use a custom Gemfile separate from the project with a different Ruby version. Notice that

Reviewer: ```suggestion
The alternative is to use a custom Gemfile separate from the project with a different Ruby version. Notice that
```

Reviewer: Ah I see, ignore.

certain functionality may be degraded or require manual configuration, since the Ruby LSP will not be able to inspect

> the project's real bundle to discover dependencies. Please, see the instructions below.

Reviewer: ```suggestion
the project's real bundle to discover dependencies. Please see the instructions below.
```

#### Using a custom Gemfile

If you are working on a project using an older version of Ruby not supported by Ruby LSP, then you may specify a
separate `Gemfile` for development tools.

**Note**: when using this, gems will not be installed automatically and neither will `ruby-lsp` upgrades.

> Create a folder to store the custom bundle outside of the project that uses the old Ruby version. Inside that folder,

Reviewer: (nit) we should really say `directory` rather than `folder`, since folder is a GUI abstraction.

> add your favorite version manager configuration to select a supported Ruby version. For example, if using `chruby`, it

Reviewer: ```suggestion
add your preferred version manager configuration to select a supported Ruby version. For example, if using `chruby`, it
```

would look like this:

```shell
# the/folder/.ruby-version

3.2.2
```

Create a `Gemfile` for development tools inside that folder.

```ruby
# the/folder/Gemfile
# frozen_string_literal: true

source "https://rubygems.org"

gem "ruby-lsp"
gem "rubocop"
```

> Don't forget to run `bundle install` inside that folder to generate a lockfile. After the folder contains the custom

Reviewer: ```suggestion
Run `bundle install` inside that folder to generate a lockfile. After the folder contains the custom
```
